### PR TITLE
[MM-59044] Add per-server permission check for screen sharing

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
We currently allow for unscrutinized use of the `getSources` API allowed a malicious server to potentially abuse this API without the user's knowledge. This PR adds a per-server permission check for screen sharing so that the user will always be asked before the screen sharing API is used.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-59044

```release-note
Fixed an issue where the screen sharing API could be used without the user's permission
```

### CODE REVIEW

File: src/main/windows/callsWidgetWindow.ts [hunk 1]
```diff
@@ -403,6 +404,11 @@ export class CallsWidgetWindow {
             }
         }
 
+        if (!await PermissionsManager.doPermissionRequest(view.webContentsId, 'screenShare', {requestingUrl: view.view.server.url.toString(), isMainFrame: false})) {
```
Comment (marianunez) [inline src/main/windows/callsWidgetWindow.ts:407]: Is this going to show when you start a screenshare or calls?

Comment (devinbinnie) [reply src/main/windows/callsWidgetWindow.ts:407]: Just a screen share.

Comment (marianunez) [reply src/main/windows/callsWidgetWindow.ts:407]: And the permission is granted once or does it ask each time?

File: src/main/permissionsManager.ts [hunk 2]
```diff
@@ -239,10 +241,12 @@ t('main.permissionsManager.checkPermission.dialog.message.media');
 t('main.permissionsManager.checkPermission.dialog.message.geolocation');
 t('main.permissionsManager.checkPermission.dialog.message.notifications');
 t('main.permissionsManager.checkPermission.dialog.message.openExternal');
+t('main.permissionsManager.checkPermission.dialog.message.screenShare');
```
Comment (marianunez) [inline src/main/permissionsManager.ts:244]: Will this now show it in the permissions settings window as well?

File: src/main/windows/callsWidgetWindow.ts (hunk 1 above)
Comment (devinbinnie) [reply src/main/windows/callsWidgetWindow.ts:407]: It's granted once, same as all of the other permissions

File: src/main/permissionsManager.ts (hunk 2 above)
Comment (devinbinnie) [reply src/main/permissionsManager.ts:244]: Yep it will

**Review by marianunez — APPROVED** [review]
Thanks @devinbinnie! LGTM 

cc// @cpoile for visibility on calls

**Review by hmhealey — APPROVED** [review]

**Review by enzowritescode — APPROVED** [review]

**Pull request merged**